src/transform/EditTransform.js

```diff
@@ -0,0 +1,34 @@
+import './EditTransform.css'
+
+const CLASS = {
+  CONTAINER: 'pagelib_edit_section_link_container',
+  LINK: 'pagelib_edit_section_link',
+  PROTECTION: { UNPROTECTED: '', PROTECTED: 'page-protected', FORBIDDEN: 'no-editing' }
+}
+
+const DATA_ATTRIBUTE = { SECTION_INDEX: 'data-id', ACTION: 'data-action' }
+const ACTION_EDIT_SECTION = 'edit_section'
+
+/**
+ * @param {!Document} document
+ * @param {!number} index The zero-based index of the section.
+ * @return {!HTMLSpanElement}
+ */
+const newEditSectionLink = (document, index) => {
+  const container = document.createElement('span')
```

Comment:
If I had to choose between decomposing the 2 functions and the naming change I'd choose the decomposing. No strong feelings about the specific naming, just things to consider - whatever you choose is fine.